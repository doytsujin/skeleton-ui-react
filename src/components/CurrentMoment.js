import React from 'react';
import moment from 'moment';

export default class CurrentMoment extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      format: props.format || 'M/D/YY h:mm A',
      now: moment()
    };
  }

  componentDidMount() {
    this.intervalId = setInterval(this.refreshCurrentMoment, 5000);
  }

  componentWillUnmount() {
    clearInterval(this.intervalId);
  }

  refreshCurrentMoment = () => {
    const now = moment();
    this.setState(() => ({ now }));
  };

  render() {
    return (
      <span>{this.state.now.format(this.state.format)}</span>
    );
  }

};