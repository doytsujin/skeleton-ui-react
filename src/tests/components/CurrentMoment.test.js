import React from 'react';
import { shallow } from 'enzyme';

import CurrentMoment from '../../components/CurrentMoment';

test('should render CurrentMoment correctly', () => {
  const wrapper =  shallow(<CurrentMoment />);
  expect(wrapper).toMatchSnapshot();
});

test('should render CurrentMoment with format correctly', () => {
  const wrapper = shallow(<CurrentMoment format='YYYY' />);
  expect(wrapper).toMatchSnapshot();
});