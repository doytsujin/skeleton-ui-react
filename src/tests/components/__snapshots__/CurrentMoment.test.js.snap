// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CurrentMoment correctly 1`] = `
<span>
  12/31/69 7:00 PM
</span>
`;

exports[`should render CurrentMoment with format correctly 1`] = `
<span>
  1969
</span>
`;
